### DISCUSSION TITLE
Make the code object-oriented

### DISCUSSION THREAD

**Discussion by Advik-B** [ThioJoe/YT-Spammer-Purge#548] [General]
I forked your repo, so that I can turn it into a GUI but I got a headache after seeing your code
try to split it into smaller files ( PyInstaller WILL STILL WORK )
this is just a friendly suggestion ( sorry if i was rude ) I get annoyed by non-OOP stuff

EDIT: FINE I WILL DO IT MYSELF

**Comment by Advik-B** [comment]
![image](https://user-images.githubusercontent.com/86160411/152564601-a7d16cec-9f4b-4dde-969c-edc81ed58224.png)
I am already working on the UI, JUST LOOK HOW BEAUTIFUL IT LOOKs

**Reply by Firecul** [reply]
I don't know whether to like this because it *has* a GUI or dislike it because of the GUI it now has.

**Reply by TechStudent10** [reply]
It looks... colorful. I would apply some extra padding on the outside of that button tho.

**Reply by Advik-B** [reply]
> 

Ok!

**Reply by Advik-B** [reply]
> It looks... colorful. I would apply some extra padding on the outside of that button tho.

The colored border is sharex after effects

**Comment by ThioJoe** [comment]
I never understood what object oriented programming even means lol

**Reply by Advik-B** [reply]
Ok, I will try to rewrite the whole thing, But I will need your help!
[here](https://www.youtube.com/watch?v=JeznW_7DlB0) is a crash course on it
because your code has SO MUCH functionality that I cannot do it myself `complement += 1`
@ThioJoe for OOP's sake please split the code Into smaller files, and make shure to leave a LOT OF COMMENTS 

( Again, sorry If i was rude, non-oop python just ticks me off 😉)

NOTE: Please DO NOT USE Tkinter for GUI, as it is NOT cross-platform, I will take care of the GUI, you do you, Thio
( I will be using PyQt5 )

**Reply by Firecul** [reply]
> please split the code Into smaller files

Should have seen it when it was just 1 file 😂

**Reply by DinhHuy2010** [reply]
ok

**Reply by TechStudent10** [reply]
@ThioJoe So basically, OOP in a nutshell is basically classes (from my limited understanding). It's just classes. Classes make projects look more organized (because they actually are).

However, @Advik-B converting a function-based program (busting out the React vocab) into an OOP-based program isn't all flowers and rainbows. To start off, you'll have to put `self.` in front of all your variables in your code. You'll have to assign them (the variables) to the class. And don't get me started on passing variables between classes. Passing variables between functions is hard enough.

I have nothing against OOP, but if you've already written a full-scale app that hundreds of people now use, converting to OOP could, and probably would cause unwanted behavior.

So unless you want to spend countless hours converting to OOP when it doesn't bring any benefit to the user, it just isn't worth it.

**Comment by Advik-B** [comment]
@ThioJoe  LOOK!

https://user-images.githubusercontent.com/86160411/152643939-c8dd890c-c96d-46ae-8f4e-ee675ef18987.mp4

NOTE: The screen glitch is NOT the code, It is OBS studio

**Reply by Firecul** [reply]
Not a critique on the code just on the video: The easier way to record this would be to have a non-offensive desktop background and turn off icons. That way you can do a screen capture and just crop as appropriate. It wouldn't glitch out as the window resizes that way.

**Reply by Kurdakov** [reply]
Do you see what is the problem?
The problem is that your GUI already limits options, like no option for multiple videos, 
it was exact reason to stick with command line interface, because fixing GUI to add some new flexible feature becomes a problem itself.

**Reply by Kurdakov** [reply]
For that reason (though I'm lazy so I do not add any code yet :) and even do not try, as I know it will take me few weeks to make as I would like and the time I would not be able to allocate ), I think better approach is to make current code to be ready for 
some GUI friendly paradigm like model - view (controller), so that command line interface becomes just view and all logic is in some easily interfaced model, then it will be possible to have variable type guis like - web gui, wxwindows, pyqt5,
I actually agree, that even simple data encapsulation in separate classes would reduce number of parameters to functions and simplify code. Maybe then - it makes sense to set priorities and have some vision given that still the program rapidly evolves, such as - to have easy to add  plug-able new filters for scanned messages, maybe some clever fsm to transition to stages of action (states), so that next transition  is track able etc...

**Reply by Advik-B** [reply]
Naa, we will do a wisard-like interface

**Comment by DinhHuy2010** [comment]
Can you extract all scan mode to each single class?

**Reply by Advik-B** [reply]
Ill try, the original version my Thio is impossible

**Reply by DinhHuy2010** [reply]
ok

**Reply by UnofficalTailsNSonicPals** [reply]
good luck

**Reply by DinhHuy2010** [reply]
where to download?

**Comment by TechStudent10** [comment]
@Advik-B converting a function-based program (busting out the React vocab) into an OOP-based program isn't all flowers and rainbows. To start off, you'll have to put `self.` in front of all your variables in your code. You'll have to assign them (the variables) to the class. And don't get me started on passing variables between classes. Passing variables between functions is hard enough.

I have nothing against OOP, but if you've already written a full-scale app that hundreds of people now use, converting to OOP could, and probably would cause unwanted behavior.

So unless you want to spend countless hours converting to OOP when it doesn't bring any benefit to the user, it just isn't worth it.

**Comment by Advik-B** [comment]
Progress is VERY SLOW

**Reply by DinhHuy2010** [reply]
I guess is your network

**Reply by Firecul** [reply]
> I guess is your network

That won't slow down his writing code (much 😉 )

**Reply by TechStudent10** [reply]
Well I mean you're probably going to StackOverflow 90% of the time so yeah network is essential.

**Reply by Firecul** [reply]
It's YT that distracts me, limited internet increases my productivity

**Reply by TechStudent10** [reply]
Makes sense

**Comment by DinhHuy2010** [comment]
what module to you use to make colorful GUI?
